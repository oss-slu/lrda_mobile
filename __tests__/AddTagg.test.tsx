import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import TagWindow from '../lib/components/tagging';


jest.mock('../lib/utils/api_calls', () => ({
  fetchCreatorName: jest.fn(() => Promise.resolve([])),
}));


beforeEach(() => {
  // Clear mocks before each test
  jest.clearAllMocks();

  // Mock console methods to avoid unnecessary log outputs in tests
  jest.spyOn(console, 'log').mockImplementation(() => {});
  jest.spyOn(console, 'error').mockImplementation(() => {});
});

afterEach(() => {
  // Restore the original console methods
  console.log.mockRestore();
  console.error.mockRestore();
});

jest.mock('@10play/tentap-editor', () => ({
  RichText: () => null,
  Toolbar: () => null,
  useEditorBridge: jest.fn(() => ({
    getHTML: jest.fn(() => ''),
    commands: {
      setContent: jest.fn(),
      focus: jest.fn(),
    },
  })),
}));

describe('TagWindowTest1', () => {
  const mockTags = ['Tag1', 'Tag2', 'Tag3'];
  const mockSetTags = jest.fn();

  it('renders without crashing', () => {
    const { toJSON } = render(<TagWindow tags={mockTags} setTags={mockSetTags} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('displays input field and tags list', () => {
    const { getByTestId, getAllByText } = render(<TagWindow tags={mockTags} setTags={mockSetTags} />);
    
    // Check if the input field exists
    expect(getByTestId('tag-input')).toBeTruthy();

    // Check if tags are displayed
    mockTags.forEach(tag => {
      expect(getAllByText(tag).length).toBeGreaterThan(0);
    });
  });
});

describe('TagWindowTest2', () => {
  const mockTags = ['Tag1', 'Tag2', 'Tag3'];
  const mockSetTags = jest.fn();

  it('handles tag deletion when swiping', async () => {
    const { getByTestId, getAllByText } = render(<TagWindow tags={mockTags} setTags={mockSetTags} />);

    // Simulate a swipe to delete the first tag
    const swipeListView = getByTestId('swipe-list');

    // Trigger row open, simulating a swipe to delete the first tag (Tag1)
    fireEvent(swipeListView, 'onRowOpen', '0');

    // Wait for the mockSetTags to be called with the updated tags
    await waitFor(() => {
      expect(mockSetTags).toHaveBeenCalledWith(['Tag2', 'Tag3']);
    });
  });
});
