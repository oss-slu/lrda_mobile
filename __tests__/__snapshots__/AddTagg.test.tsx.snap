// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagWindowTest1 renders without crashing 1`] = `
<View
  style={
    {
      "marginBottom": 10,
      "minHeight": 100,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholder="Your Tag Here"
    style={
      {
        "backgroundColor": "white",
        "borderBottomWidth": 2,
        "borderRadius": 30,
        "fontSize": 18,
        "fontWeight": "500",
        "height": 40,
        "justifyContent": "center",
        "textAlign": "center",
        "width": "100%",
      }
    }
    testID="tag-input"
    value=""
  />
  <RCTScrollView
    closeOnRowBeginSwipe={false}
    closeOnRowOpen={true}
    closeOnRowPress={true}
    closeOnScroll={true}
    data={
      [
        {
          "key": "0",
          "tag": "Tag1",
        },
        {
          "key": "1",
          "tag": "Tag2",
        },
        {
          "key": "2",
          "tag": "Tag3",
        },
      ]
    }
    directionalDistanceChangeThreshold={2}
    disableHiddenLayoutCalculation={false}
    disableLeftSwipe={false}
    disableRightSwipe={false}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    leftActivationValue={160}
    leftOpenValue={75}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRowOpen={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    previewFirstRow={false}
    previewRepeat={false}
    previewRepeatDelay={1000}
    recalculateHiddenLayout={false}
    removeClippedSubviews={false}
    renderHiddenItem={[Function]}
    renderItem={[Function]}
    rightActivationValue={-160}
    rightOpenValue={-75}
    scrollEnabled={false}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    stopLeftSwipe={175}
    stopRightSwipe={-175}
    swipeToClosePercent={50}
    swipeToOpenPercent={50}
    swipeToOpenVelocityContribution={0}
    testID="swipe-list"
    useAnimatedList={false}
    useNativeDriver={true}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={{}}
        >
          <View
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 1,
                },
                {
                  "height": 0,
                  "width": 0,
                },
              ]
            }
          >
            <View
              leftActionActivated={false}
              leftActionState={false}
              rightActionActivated={false}
              rightActionState={false}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 15,
                  "paddingVertical": 1,
                }
              }
              swipeAnimatedValue={0}
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="delete-action-0"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#111111",
                        "fontSize": 24,
                      },
                      {
                        "alignSelf": "center",
                      },
                      {
                        "fontFamily": "ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  î»·
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            manipulationModes={
              [
                "translateX",
              ]
            }
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              {
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "zIndex": 2,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={false}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "fontSize": 18,
                      "fontWeight": "500",
                    }
                  }
                >
                  Tag1
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={{}}
        >
          <View
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 1,
                },
                {
                  "height": 0,
                  "width": 0,
                },
              ]
            }
          >
            <View
              leftActionActivated={false}
              leftActionState={false}
              rightActionActivated={false}
              rightActionState={false}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 15,
                  "paddingVertical": 1,
                }
              }
              swipeAnimatedValue={0}
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="delete-action-1"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#111111",
                        "fontSize": 24,
                      },
                      {
                        "alignSelf": "center",
                      },
                      {
                        "fontFamily": "ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  î»·
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            manipulationModes={
              [
                "translateX",
              ]
            }
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              {
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "zIndex": 2,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={false}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "fontSize": 18,
                      "fontWeight": "500",
                    }
                  }
                >
                  Tag2
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={{}}
        >
          <View
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 1,
                },
                {
                  "height": 0,
                  "width": 0,
                },
              ]
            }
          >
            <View
              leftActionActivated={false}
              leftActionState={false}
              rightActionActivated={false}
              rightActionState={false}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "red",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 15,
                  "paddingVertical": 1,
                }
              }
              swipeAnimatedValue={0}
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
                testID="delete-action-2"
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#111111",
                        "fontSize": 24,
                      },
                      {
                        "alignSelf": "center",
                      },
                      {
                        "fontFamily": "ionicons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  î»·
                </Text>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            manipulationModes={
              [
                "translateX",
              ]
            }
            onLayout={[Function]}
            onMoveShouldSetResponder={[Function]}
            onMoveShouldSetResponderCapture={[Function]}
            onResponderEnd={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderReject={[Function]}
            onResponderRelease={[Function]}
            onResponderStart={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            onStartShouldSetResponderCapture={[Function]}
            style={
              {
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "zIndex": 2,
              }
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={false}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingVertical": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "fontSize": 18,
                      "fontWeight": "500",
                    }
                  }
                >
                  Tag3
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
